ci-metrics/ci_listener.py

```diff
@@ -266,88 +270,45 @@ def handle_tests(self, key, value, retried=False):
-                        self.message_out["%s_total_tests_failed" %
-                            executor] = int(tester["failed"]) + \
-                            self.message_out["%s_total_tests_failed" % executor]
-                else:
-                    self.message_out["%s_total_tests_failed" %
-                        executor] = int(tester["failed"])
-                if "total_tests_failed" in self.message_out.keys():
-                    self.message_out["total_tests_failed"] = \
-                        int(tester["failed"]) + \
-                        self.message_out["total_tests_failed"]
-                else:
-                    self.message_out["total_tests_failed"] = \
-                        int(tester["failed"])
-                if "total_time_secs" in self.message_out.keys():
-                    self.message_out["total_time_secs"] = \
-                        max(int(time_end - time_start), 0) + \
-                        self.message_out["total_time_secs"]
-                else:
-                    self.message_out["total_time_secs"] = \
-                        max(int(time_end - time_start), 0)
-                if "total_time_mins" in self.message_out.keys():
-                    self.message_out["total_time_mins"] = \
-                        max(int(time_end - time_start), 0)/60.0 + \
-                        self.message_out["total_time_mins"]
-                else:
-                    self.message_out["total_time_mins"] = \
-                        max(int(time_end - time_start), 0)/60.0
-                if "total_time_hrs" in self.message_out.keys():
-                    self.message_out["total_time_hrs"] = \
-                        max(int(time_end - time_start), 0)/3600.0 + \
-                        self.message_out["total_time_hrs"]
-                else:
-                    self.message_out["total_time_hrs"] = \
-                        max(int(time_end - time_start), 0)/3600.0
+                supplements = [ \
+                    ["%s_total_tests_exec" % executor, execed],
+                    ["total_tests_exec", execed],
+                    ["%s_total_tests_failed" % executor, failed],
+                    ["total_tests_failed", failed],
+                    ["total_time_secs", seconds],
+                    ["total_time_mins", seconds/60.0],
+                    ["total_time_hrs", seconds/3600.0]] 
+                for key, addition in supplements:
+                    self.message_key_addition(key, addition)
         return True
 
     def handle_recipients(self, key, value, retried=False):
-        recipient_set = Set([])
         # Add old recipients to the set
-        if "recipients" in self.message_out.keys():
-            for userid in self.message_out["recipients"].split(','):
-                recipient_set.add(userid[1].strip())
+        recipient_set = Set(map(lambda x: x.strip(), 
+            str(self.message_out.get("recipients")).split(",")))
+        recipient_set.discard("None")
```

Comment:
Hmm.. that discard is annoying.
How about this?

`recipient_set = Set()
if "recipients" in self.message_out:
    recipient_set.update(map(lambda x: x.strip(), message_out.get("recipients","").split(",")))
`